pkg/daemon/ceph/osd/agent_test.go

```diff
@@ -40,6 +41,24 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func init() {
```

Comment:
did you intend `init()` to be called before each test method? and cleanup when the test is done? It looks like it will immediately delete the temp folders instead of at the end of the test. If you need some init and cleanup code for a unit test, you can use a Suite, like what is used for the integration tests.

Reply:
Hmm. That's not exactly what I intended, but that sounds like the better way.

Reply:
FYI, I plan to poke at this tomorrow. I have been a bit sidetracked today.